version: '2.1'

volumes:
   db-backups:
   geoserver-data:
   geo-db-data:

services:
  db:
    image: kartoza/postgis:11.0-2.5
    volumes:
        - geo-db-data:/var/lib/postgresql
    ports:
      - "25434:5432"
    env_file:
    - geoserver/docker-env/db.env
    restart: on-failure
    healthcheck:
      test: "exit 0"

  geoserver:
    image: kartoza/geoserver:2.15.2
    build:
      context: ./geoserver
      dockerfile: ./Dockerfile
    volumes:
      - geoserver-data:/opt/geoserver/data_dir
      - ./geoserver/.data:/opt/data
    ports:
      - "8600:8080"
    restart: on-failure
    env_file:
      - geoserver/docker-env/geoserver.env
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: curl --fail -s http://localhost:8080/ || exit 1
      interval: 1m30s
      timeout: 10s
      retries: 3


  db-dbbackups:
    image: kartoza/pg-backup:11.0
    volumes:
    - db-backups:/backups
    env_file:
    - geoserver/docker-env/db-backups.env
    restart: on-failure
    depends_on:
      db:
        condition: service_healthy

  adminer:
    image: adminer
    # restart: always
    ports:
      - 8700:8080

  ged:
    image: ged
    volumes:
      - ./ged:/opt/app
      - ./:/opt/root
    build:
      context: ./ged
      dockerfile: ./Dockerfile
    ports:
      - 8888:7888
      - 8800:9500
      - 9630:9630
      - 8801:8801
      - 8899:8899
    # command: bash c main
    command: tail -f /dev/null
    environment:
      - STAGE=dev
    stdin_open: true
    tty: true
